src/renderer.ts

```diff
@@ -203,16 +214,436 @@ export class DefaultCellRenderer implements ICellRenderer {
+    });
+
+
+    var bits = [];
+    var min = col.desc.sdomain[0];
+    var max = col.desc.sdomain[1];
+    var mincolor = col.desc.colorrange[0];
+    var maxcolor = col.desc.colorrange[1];
+    var barheight;
+    var threshold = 0;
+    var scale = d3.scale.linear();
+
+    var $rects = $rows_enter.selectAll('rect').data(function (d, i) {
+      var value = col.getValue(d);
+      var data = value;
+      bits.push(data.length);
+
+      barheight = context.rowHeight(i);
+      return (data);
+    });
+
+    scale = (min < 0) ? (scale.domain([min, max]).range([0, barheight / 2])) : (scale.domain([min, max]).range([0, barheight]));
+    var color: any = d3.scale.linear<number,string>();
+    color.domain([min, 0, max]).range([mincolor, 'white', maxcolor]);
+
+    $rects.enter().append('rect');
+    $rects.attr({
+      'data-index': (d, i) => i,
+      'width': (col.getWidth() / d3.max(bits)),
+      'height': (d: any) => (d < threshold) ? (barheight / 2 - scale(d)) : scale(d),
+      'x': function (d) {
+
+        var prev = this.previousSibling; // One previous step information.
+        return (prev === null) ? 0 : parseFloat(d3.select(prev).attr('x')) + parseFloat(d3.select(prev).attr('width'));
+      },
+      'y': function (d: any, i) {
+        if (min < 0) {
+          return (d < threshold) ? (context.rowHeight(i) / 2) : context.rowHeight(i) / 2 - scale(d);   // For positive and negative value
+        } else {
+          return context.rowHeight(i) - scale(d);
+        }
+      },
+      'fill': color
+    });
+
+
+    context.animated($rows).attr({
+      transform: (d, i) => 'translate(' + context.cellX(i) + ',' + context.cellY(i) + ')'
+    });
+    $rows.exit().remove();
+  }
+
+}
+
+class BoxplotCellRenderer extends DefaultCellRenderer {
+  render($col: d3.Selection<any>, col: model.MyColumn, rows: any[], context: IRenderContext) {
+
+
+
+    var maxarr = [];
+    var minarr = [];
+    var q1arr = [];
+    var q3arr = [];
+    var medarr = [];
+
+
+    function getPercentile(data, percentile) {
+      data.sort(numSort);
+      var index = (percentile / 100) * data.length;
+      var result;
+      if (Math.floor(index) === index) {
+        result = (data[(index - 1)] + data[index]) / 2;
+      } else {
+        result = data[Math.floor(index)];
+      }
+      return result;
+    }
+
+//because .sort() doesn't sort numbers correctly
+    function numSort(a, b) {
+      return a - b;
+    }
+
+
+    rows.forEach(function (d, i) {
+
+      var data = col.getValue(d);
+
+      var m1 = Math.min.apply(Math, data);
+      var m2 = Math.max.apply(Math, data);
+      minarr.push(m1);
+
+      maxarr.push(m2);
+
+      q1arr.push(getPercentile(data, 25));
+      medarr.push(getPercentile(data, 50));
+      q3arr.push(getPercentile(data, 75));
+
+    });
+
+
+
+    var scale = d3.scale.linear().domain([d3.min(minarr), d3.max(maxarr)]).range([0, col.getWidth()]); // Constraint the window width
```

Comment:
using `d3.min(minarr)` ... it is based just on the currently provided data. That might be a subset of all data.